src/Sylius/Bundle/ApiBundle/Controller/DeleteOrderItemAction.php

```diff
@@ -34,7 +34,11 @@ public function __invoke(Request $request): Response
         $orderItemId = $request->attributes->get('itemId');
         $tokenValue = $request->attributes->get('tokenValue');
 
-        $orderItem = $this->orderItemRepository->findOneByIdAndCartTokenValue($orderItemId, $tokenValue);
+        try {
+            $orderItem = $this->orderItemRepository->findOneByIdAndCartTokenValue($orderItemId, $tokenValue);
+        } catch (\Exception) {
+            throw new OrderItemNotFoundException();
+        }
```

Comment:
Btw, in such cases, always pass the previous exception as a `previous` parameter. It will keep the stack trace before the throwing `OrderItemNotFoundException`.

Reply:
In fact, the problem is how PostgreSQL handles int/string identifiers

<img width="1255" alt="image" src="https://github.com/Sylius/Sylius/assets/40125720/7037c7e5-c08b-4cfb-9d68-791221547645">

Reply:
What's the exact error 🤔? Aren't we able to fix it? If no, let's catch `DriverException`, not the generic one.

Reply:
In this case, I was able to handle the exception, but there are places where the command is simply dispatched and there is no place to catch the exception. I also found an issue regarding this topic:

https://github.com/api-platform/core/issues/292

Reply:
I never liked that the API uses the `id` for order items and units. I would like it to use the product variant code.
Nevertheless, catching the generic exception here will just hide unexpected issues and we should try to avoid that.

Another thing that is not clear to me is why this action tries to get the order item when the handler already does that and throws an exception when not found.

Reply:
I redesigned it to capture this PostgreSQL edge case globally:
https://github.com/Sylius/Sylius/pull/15064/files#diff-2fc6473482e2d7142d8007592f5fe2ae76fc9db4c157ca4bb8eb3b6feefd7a47